src/Common/src/Interop/Unix/Interop.Errors.cs

```diff
@@ -1,48 +1,166 @@
+    }
+
+    // Represents a platform-agnostic Error and underlying platform-specific errno
+    internal struct ErrorInfo
+    {
+        internal readonly Error Error;
+        internal readonly int Errno;
+
+        internal ErrorInfo(Error error)
+        {
+            Error = error;
+            Errno = Interop.System.ConvertErrorPalToPlatform(error);
+        }
+
+        internal ErrorInfo(int errno)
+        {
+            Error = Interop.System.ConvertErrorPlatformToPal(errno);
+            Errno = errno;
+        }
+
+        internal ErrorInfo(Error error, int errno)
+        {
+            Error = error;
+            Errno = errno;
+        }
+
+        internal string GetErrorMessage()
+        {
+            return Interop.System.StrError(Errno);
+        }
+    }
+
+    internal partial class System
     {
-        // These values were defined in:
-        // include/asm-generic/errno-base.h
-        // /usr/include/sys/errno.h
-
-        internal const int EPERM = 1;
-        internal const int ENOENT = 2;
-        internal const int ESRCH = 3;
-        internal const int EINTR = 4;
-        internal const int EIO = 5;
-        internal const int ENXIO = 6;
-        internal const int E2BIG = 7;
-        internal const int ENOEXEC = 8;
-        internal const int EBADF = 9;
-        internal const int ECHILD = 10;
-
-        internal const int ENOMEM = 12;
-        internal const int EACCES = 13;
-        internal const int EFAULT = 14;
-        internal const int ENOTBLK = 15;
-        internal const int EBUSY = 16;
-        internal const int EEXIST = 17;
-        internal const int EXDEV = 18;
-        internal const int ENODEV = 19;
-        internal const int ENOTDIR = 20;
-        internal const int EISDIR = 21;
-        internal const int EINVAL = 22;
-        internal const int ENFILE = 23;
-        internal const int EMFILE = 24;
-        internal const int ENOTTY = 25;
-        internal const int ETXTBSY = 26;
-        internal const int EFBIG = 27;
-        internal const int ENOSPC = 28;
-        internal const int ESPIPE = 29;
-        internal const int EROFS = 30;
-        internal const int EMLINK = 31;
-        internal const int EPIPE = 32;
-        internal const int EDOM = 33;
-        internal const int ERANGE = 34;
+        internal static Error GetLastError()
+        {
+            return ConvertErrorPlatformToPal(Marshal.GetLastWin32Error());
+        }
+
+        internal static ErrorInfo GetLastErrorInfo()
+        {
+            return new ErrorInfo(Marshal.GetLastWin32Error());
+        }
+
+        internal static unsafe string StrError(int platformErrno)
+        {
+            const int maxErrorMessageLength = 1024; // length long enough for most any Unix error
+            byte* buffer = stackalloc byte[maxErrorMessageLength];
+            IntPtr message = StrErrorR(platformErrno, buffer, maxErrorMessageLength);
+            return Marshal.PtrToStringAnsi(message);
```

Comment:
I'm glad all that horrible gook I had for choosing between strerror_r's in managed goes away :)